src/main/java/org/peergos/net/HttpProxyHandler.java

```diff
@@ -12,20 +12,29 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.logging.Logger;
 import java.util.stream.Collectors;
 
 public class HttpProxyHandler extends Handler {
-	private static final Logger LOG = Logging.LOG();
+    private static final Logger LOG = Logging.LOG();
 
     private static final boolean LOGGING = true;
 
     private final HttpProxyService service;
+    private final Function<String, Multihash> peerIdToMultiHashConverter;
 
     private static final String HTTP_REQUEST = "/http/";
+    private static final Function<String, Multihash> defaultPeerIdToMultiHashConverter = (peerIdStr) ->
+         Multihash.deserialize(PeerId.fromBase58(peerIdStr).getBytes());
```

Comment:
Any reason not to just make this a static function? Can also just directly decode the base58, don't need to go via PeerId.